src/java.base/share/classes/jdk/internal/module/ModuleInfo.java

```diff
@@ -280,8 +282,15 @@ private Attributes doRead(DataInput in) throws IOException {
                     } else {
                         in.skipBytes(length);
                     }
+            }
 
+            long newPosition = in.count();
+            if (newPosition != (initialPosition + length)) {
```

Comment:
should this be:

```
if ((newPosition - intialPosition) != length) {
```

just to make it clear that no overflow can happen?

Reply:
The attribute_length is a u4 so shouldn't arise but you are probably right that it would be clearer to reorganize it as you suggest.